### ISSUE TITLE
"Invalid 'main' field" warning

### ISSUE THREAD

**Issue by Piedone** [Azure/webapps-deploy#270]
I get the following warning when running the action:

```
(node:4284) [DEP0128] DeprecationWarning: Invalid 'main' field in '/home/runner/work/_actions/azure/webapps-deploy/016bdd3f9b7cec60310bcf9da98f671628795644/node_modules/actions-secret-parser/package.json' of 'lib/index.js'. Please either fix that or report it to the module author
```

**Comment by Nicolai-Bak** [comment]
Had the same issue when trying to deplay an ASP.NET 6 api. Problem was solved by updating `actions/checkout@v2` and `actions/setup-dotnet@v1` to `actions/checkout@v3` and `actions/setup-dotnet@v3`

**Comment by oshihirii** [comment]
Had same issue now when using:
```
uses: actions/checkout@v3
uses: actions/setup-node@v3
uses: actions/upload-artifact@v3
uses: actions/download-artifact@v3
```  
> (node:2310) [DEP0128] DeprecationWarning: Invalid 'main' field in '/home/runner/work/_actions/azure/webapps-deploy/v2/node_modules/actions-secret-parser/package.json' of 'lib/index.js'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
Package deployment using ZIP Deploy initiated.

My full workflow file is below incase it helps with troubleshooting:
```
# Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
# More GitHub Actions for Azure: https://github.com/Azure/actions

name: Build and deploy Node.js app to Azure Web App - <app-name-here>

on:
  push:
    branches:
      - main
  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3

      - name: Set up Node.js version
        uses: actions/setup-node@v3
        with:
          node-version: '18.x'

      - name: npm install, build, and test
        run: |
          npm install
          npm run build --if-present
          npm run test --if-present

      - name: Upload artifact for deployment job
        uses: actions/upload-artifact@v3
        with:
          name: node-app
          path: .

  deploy:
    runs-on: ubuntu-latest
    needs: build
    environment:
      name: 'Production'
      url: ${{ steps.deploy-to-webapp.outputs.webapp-url }}

    steps:
      - name: Download artifact from build job
        uses: actions/download-artifact@v3
        with:
          name: node-app

      - name: 'Deploy to Azure Web App'
        id: deploy-to-webapp
        uses: azure/webapps-deploy@v2
        with:
          app-name: '<app-name-here>'
          slot-name: 'Production'
          publish-profile: ${{ secrets.AZUREAPPSERVICE_PUBLISHPROFILE_<********> }}
          package: .
```

**Comment by shpraka** [comment]
This has been fixed as part of #280, will be rolled out in next release.
